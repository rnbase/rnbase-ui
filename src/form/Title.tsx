import React from 'react';
import { StyleSheet, Text, TextProps } from 'react-native';

import { Theme, Themed, withTheme } from '../theming';

interface TitleProps extends TextProps {
  children: React.ReactNode;
}

const Title: React.FC<Themed<typeof createStyleSheet, TitleProps>> = ({
  theme: { styles },
  children,
  style,
  ...props
}) => (
  <Text {...props} style={[styles.root, style]} numberOfLines={1}>
    {children}
  </Text>
);

const createStyleSheet = ({ colors, fonts }: Theme) =>
  StyleSheet.create({
    root: {
      ...fonts.bold,
      margin: 15,
      height: 13,
      fontSize: 13,
      lineHeight: 13,
      marginBottom: -10,
      color: colors.black,
      textTransform: 'uppercase',
      backgroundColor: colors.transparent,
    },
  });

export default withTheme(Title, createStyleSheet, 'Form.Title');
