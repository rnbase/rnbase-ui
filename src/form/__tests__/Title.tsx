import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

import Title from '../Title';

type Props = React.ComponentProps<typeof Title>;

/**
 * Under test
 */
const createElement = (props: Partial<Props>) => <Title {...props}>title</Title>;

const createRenderer = (props: Partial<Props>) => TestRenderer.create(createElement(props));

it('should render normally', () => {
  expect(createRenderer({})).toMatchSnapshot();
});

it('should render with custom styles', () => {
  expect(createRenderer({ style: { color: 'red' } })).toMatchSnapshot();
});
