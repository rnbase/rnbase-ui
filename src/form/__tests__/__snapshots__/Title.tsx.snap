// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render normally 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 13,
        "fontWeight": "700",
        "height": 13,
        "includeFontPadding": false,
        "lineHeight": 13,
        "margin": 15,
        "marginBottom": -10,
        "textAlignVertical": "center",
        "textTransform": "uppercase",
      },
      undefined,
    ]
  }
>
  title
</Text>
`;

exports[`should render with custom styles 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 13,
        "fontWeight": "700",
        "height": 13,
        "includeFontPadding": false,
        "lineHeight": 13,
        "margin": 15,
        "marginBottom": -10,
        "textAlignVertical": "center",
        "textTransform": "uppercase",
      },
      Object {
        "color": "red",
      },
    ]
  }
>
  title
</Text>
`;
